arrivals/src/main/scala/com/pagerduty/arrivals/impl/filter/FilterDirectives.scala

```diff
@@ -0,0 +1,46 @@
+package com.pagerduty.arrivals.impl.filter
+
+import akka.http.scaladsl.server.{Directive, Directive0, Directive1, RouteResult}
+import com.pagerduty.arrivals.api.filter._
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.RouteResult.Complete
+
+import scala.concurrent.Future
+
+object FilterDirectives {
+
+  def filterRequest[RequestData](requestFilter: RequestFilter[RequestData], requestData: RequestData): Directive0 = {
+    extractRequestFilterResult(requestFilter, requestData).flatMap {
+      case Right(filteredRequest) => mapRequest(_ => filteredRequest)
+      case Left(response)         => complete(response)
+    }
+  }
+
+  private def extractRequestFilterResult[RequestData](
+      requestFilter: RequestFilter[RequestData],
+      requestData: RequestData
+    ): Directive1[SyncRequestFilterOutput] = {
+    Directive { inner => ctx =>
+      implicit val ec = ctx.executionContext
+
+      requestFilter(ctx.request, requestData).flatMap { result =>
+        inner(Tuple1(result))(ctx)
+      }
+    }
+  }
+
+  def filterResponse[RequestData](responseFilter: ResponseFilter[RequestData], requestData: RequestData): Directive0 = {
+    Directive { inner ⇒ ctx ⇒
```

Comment:
these `=>` are rendering funny for me... are these the right chars???

Reply:
Ah, I believe this is copy-paste from the Akka source. The unicode arrow is perfectly acceptable as a substitute for `=>` as far as the compiler is concerned, but I'm not a big fan of having characters that can't be typed without special setup. I'll replace them.